### ISSUE TITLE
[REST API] Allow user access to UUID resource

### ISSUE THREAD

**Issue by jsetton** [openhab/openhab-core#2350]
Related to the discussion in #1570, I am currently working on refactoring the Alexa skill code. Part of it is the need to generate an [Alexa custom identifier](https://developer.amazon.com/en-US/docs/alexa/device-apis/alexa-discovery-objects.html#additionalattributes-object) globally unique across user accounts.

Having the ability to use the server UUID as part of this identifier would be ideal. Unfortunately, this data point requires admin access currently. Is there any downside allowing user access to the UUID resource?

**Comment by digitaldan** [comment]
hmm, we have the UUID as part of the registration for openHAB cloud.  Could we expose something there just for the skill and then for those not using the cloud service, make the UUID a optional parameter to self hosted skills targeting an OH instance directly ?

**Comment by jsetton** [comment]
We could. I was trying to limit the number of ways on how to determine that data point since that resource is available for OH2 instances.

If it's a problem to allow user access to the UUID resource, we could just have the skill generate custom identifier for OH2 instances and only focus having the planned Alexa binding handle that part for OH3 instances.

**Comment by digitaldan** [comment]
Yeah, reading #1570 sounds like its going to be a tough argument to make, so i would go with an alternative.  I was also thinking if we needed to, it may be possible to have myopenhab as part of the authentication flow for alexa, forward the user to log into their openhab after logging into the cloud, then request an api token, but this would be a lot more work if its even possible, and require a double login, so maybe not the best solution.  I'm not sure if somehow we could have the  cloud binding grab a token, but i worry we start degrading the security at this point.    

So long story short, your alternative sounds appealing ;-)

**Comment by jsetton** [comment]
> Yeah, reading #1570 sounds like its going to be a tough argument to make

My understanding is that discussion was more about UUID potentially being added to the root resource then needing to be secured. And ultimately, the consensus was to keep it separate since it has its own resource endpoint.

> I was also thinking if we needed to, it may be possible to have myopenhab as part of the authentication flow for alexa, forward the user to log into their openhab after logging into the cloud, then request an api token, but this would be a lot more work if its even possible, and require a double login, so maybe not the best solution.

I agree. Too many authentications for little reward. Once the planned Alexa binding is implemented, there will be no need for it. This is more to keep consistency between skill only and skill + binding setup in the future.

> So long story short, your alternative sounds appealing ;-)

My only concern is there will be couple OH3 versions before the planned binding is actually available that wouldn't support that custom identifier. It's not the end of the world as this property is just recommended and not required.

**Comment by kaikreuzer** [comment]
I do not see any strong reason not to expose the UUID to users - after all, it is no secret, but only a unique identifier.
The only risk might be that someone else could claim to be the same instance. But in all such cases, there will need to be some other authentication mechanism in place (like with the secret for myopenHAB).
